lib/virthck.rb

```diff
@@ -15,11 +15,7 @@ def initialize(project)
   end
 
   def assign_id
-    @id = @id_gen.allocate
-    if @id.negative?
-      @logger.fatal('No available ID, wait for a test session to end')
-      exit 1
-    end
+    @id = @id_gen.allocate while @id.negative?
```

Comment:
I think we need to wait for a while here. Otherwise, we just are in the loop and blowing up the log. Let's say next retry in 20 seconds (use constant or create config value)

Reply:
Fixed it with proper logging too